File: 프로그래머스/lv2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    int total = (brown +4) / 2;
    
    for(int i = 3 ; i <= total/2 ; i++){
        if(i * (total-i) - brown == yellow){
            answer.insert(answer.begin(), total-i);
            answer.insert(answer.begin()+1, i);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        
        for (int i=1 ; i<=(yellow/2)+1 ; i++) {
            if (yellow % i != 0) continue;
            if (brown == (i + (yellow/i) + 2)*2){
                answer = new int[]{(yellow/i)+2, i+2};
                break;
            }
        }
        return answer;
    }
}